workflow/asynchronous-in-js/callback.js

```diff
@@ -1,107 +1,95 @@
-function loadOneByOneCallback() {
-  function loadImage(url, callback, whatToCall) {
+const loadOneByOneCallback = () => {
+  const loadImage = (url, whatToCall) => {
     createAndAppendSpinnerDiv(createNewId());
 
     const xhr = new XMLHttpRequest();
     xhr.open('GET', url);
-    xhr.onload = function () {
-      const src = JSON.parse(xhr.response)[0].url;
-      callback(null, src, whatToCall);
+    xhr.onload = () => {
+      const src = JSON.parse(xhr.response || null)?.[0]?.url;
+      createImage(null, src, whatToCall);
     };
-    xhr.onerror = function () {
-      callback(error);
+    xhr.onerror = () => {
+      createImage(error);
     };
     xhr.send();
-  }
+  };
 
-  function createImage(error, src, collLoadImage) {
+  const createImage = (error, src, cb) => {
```

Comment:
I didn't notice this before, but not good approach to pass error as 1st argument to function main purpose of which not to handle errors. 
There're couple ways here: 
- pass error as last argument. But in case of error you need to invoke function like that - `createImage(null, null, err)`
- not to pass any arguments in case of error
- create separate function for error (it's more convenient for us here, because we separate creating image and showing error, and you can reuse this function for other methods)